presto-spark-base/src/test/java/com/facebook/presto/spark/TestPrestoSparkQueryRunner.java

```diff
@@ -787,6 +789,23 @@ public void testTimeouts()
         assertQueryFails(queryMaxExecutionTimeLimitSession, longRunningCrossJoin, "Query exceeded maximum time limit of 2.00s");
     }
 
+    @Test
+    public void testServiceWaitTimeMetrics()
+    {
+        // Expected run time for this query is ~30s. We will set `dummyServiceWaitTime` as 60s.
+        // The timeout logic will set the timeout for the query as 65s (Actual timeout + waitTime)
+        // and the query should succeed. This is a bit hacky way to check whether service waitTime
+        // is added to the deadline time while submitting jobs
+        System.setProperty(DUMMY_SERVICE_WAIT_TIME, String.valueOf(TimeUnit.SECONDS.toMillis(60)));
```

Comment:
I would try to avoid setting global system properties. What you can do instead, you can expose the `PrestoSparkDummyServiceWaitTimeMetrics` in the `PrestoSparkQueryRunner` and use a setter to update the value. 

Also keep in mind that tests are run in parallel. Updating the wait time in a bad moment (when the `testTimeouts`) may cause test failures. To make sure that `testTimeouts` and `testServiceWaitTimeMetrics` don't run at the same time I would consider merging them into a single test method.